manifests/component/custom_config.pp

```diff
@@ -3,10 +3,10 @@
 # with any of the shipped models.  
 #
 define rsyslog::component::custom_config (
-  Integer $priority,
-  String  $target,
-  String  $confdir,
   String  $content,
+  Integer $priority = $rsyslog::custom_priority,
+  String  $target = "${name}.conf",
+  String  $confdir = $rsyslog::confdir,
```

Comment:
unless you are directly calling `rsyslog::component::custom_config`, then these defaults would never be loaded. As this module uses Module Data to load defaults instead of a `params.pp`, then any defaults set within code would only be loaded if the module data didn't contain those defaults.

Additionally, when this defined type is called through the `rsyslog::config::custom_config` class, those three parameters are inherited from the class and, again, the defaults wouldn't apply.

Finally, we generally don't write modules in a way that implements other module features directly within a module. Instead, that type of code is a better fit for the profile that you are using to apply rsyslog code.